text/0001-script-based-tokenizer.md

```diff
@@ -0,0 +1,414 @@
+
+### Motivation
+
+We want to provide our users with an always improved searching experience. For that matter, it is critical for us to improve the performance of our tokenizer, and to provide better support for multilingual tokenization.
+
+### Prior Art and R&D
+
+**tokenization:**
+- > **[unicode-segmentation](https://github.com/unicode-rs/unicode-segmentation):**
+  > tokenizer which follow the [Standard Annex #29: Unicode Text Segmentation](http://www.unicode.org/reports/tr29/),
+  > this tokenizer seems promising for Latin scripts.
+- > **[Jieba](https://github.com/messense/jieba-rs):**
+  > tokenizer specialized in Chinese languages
+- > **[Lindera](https://github.com/lindera-morphology/lindera):**
+  > Japanese and Korean
+
+**lang/script detection:**
+- > **[whatlang](https://github.com/greyblake/whatlang-rs):**
+  > whatlang is able to detect script or/and language in a text,
+  > language detection is low but script is acceptable.
+  > note: Sonic also uses whatlang to peform the tokenization, it could be interesting to checkout how they do it.
+- > **toku (@qdequele):**
+  > in a R&D project, @qdequele was able to detect language based on stop word distribution in a text.
+  > If, in a latin script, there is lot of French stop words then the text language is probably french.
+
+**other solution that advertise multilingual support:**
+- Sonic uses whatlang to peform the tokenization, it could be interesting to checkout how they do it: https://github.com/valeriansaliou/sonic/tree/master/src/lexer
+	- Sonic uses whatlang to detect the languages, but don't acutally seems to use it to segment the text. It simply uses unicode segmentation, I can't really explain what they actually do with the language information.
+- tantivy advertise good multilingual support: https://github.com/tantivy-search/tantivy/tree/main/src/tokenizer
+	- Tantivy is similar to elastic in the sense that you can setup a custom text analyzer. The difference is that it is only made of `tokenizer` -> `token_filter`. Tantivy also provides a colleciton of tokenizer to choose from. Token are rather simple and do not contain any metadata, except for it's position.
+- How elastic search handle it: https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-tokenizers.html:
+	- Elastic proposed to create custom text analyzer. A text analyzer is a pipelined text processor with the following components: `char_filter` -> `tokenizer` -> `token_filter`. There are multiple diffferent tokenizer that can be chosed, depending on the usecase. This is a bit complicated, but I also think that advanced users should be able to chose the tokenizer they want. (default behaviour is us guessing what's the best tokenizer to use.)
```

Comment:
"I also think that advanced users should be able to choose the tokenizer they want." -> Should we use "I" in the spec? I will not be able to see the author of the spec once it's merged, right?
Maybe precise who is "I" here.

Reply:
this is a draft for now, it's still discussions we are having between us. It will eventually converge to something cleaner. I will put this is draft.

Reply:
Ok, thanks! I understand this is not the final paper now. Sorry for the useless correction then!

Reply:
no worries :)